### PR TITLE
Fix DATAWriter, CRD, PQR, and PDBQT compressed file writing

### LINKED ISSUE #4159: When trying to write a compressed data file, an uncompressed file is created with '.data.data' extension
## Expected behavior ##

When providing a LAMMPS data filename with a compressed extension, such as "test.data.bz2" or "test.data.gz" to mda.Writer, the latter should write a compressed file with the corresponding extension. 

## Actual behavior ##

In my environments, using both the development branch and the version from pip, the data files are written under the name "test.data.data" and are not compressed.

## Code to reproduce the behavior ##

Here is a simple script to write a compressed LAMMPS data file. All the attributes are set just to get the simplest datafile written without an error.

```
python3
import MDAnalysis as mda
import numpy as np
u = mda.Universe.empty(1, trajectory = True)
u.add_TopologyAttr('type')
u.add_TopologyAttr('mass')
atom = mda.core.groups.Atom(u = u, ix = 0)
atom.position = np.array([0., 0., 0.])
u.atoms = mda.AtomGroup([atom,])
u.add_bonds([])
u.add_angles([])
u.add_dihedrals([])
u.add_impropers([])
u.atoms.types = ['1',]
u.atoms.masses = [1.,]
u.dimensions = [1, 1, 1, 90, 90, 90]
with mda.Writer("test.data.bz2", n_atoms = 1) as w:
	w.write(u.atoms)
```

## Current version of MDAnalysis ##

- MDAnalysis 2.4.2
- Python 3.11.2
- Ubuntu 23.04

### PR DESCRIPTION
**Pull request by IAlibay**
Whislt this direclty Fixes #4159 it also opens a big can of worms...

## TODO?
As far as I can tell, the following file types also misuse `utils.filename` leading to this problem
- [x] CRD
- [x] PQR
- [x] PDBQT

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

----
:books: Documentation preview :books:: https://mdanalysis--4163.org.readthedocs.build/en/4163/

### CODE REVIEW

**Comment by IAlibay** [comment]
Ok this should be good to go.

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 1]
```diff
@@ -220,8 +220,13 @@ def test_molecule_tag(self, LAMMPSDATAWriter_molecule_tag):
                      err_msg="resids different after writing",)
 
 
-def test_datawriter_universe(tmpdir):
-    fn = str(tmpdir.join('out.data'))
+@pytest.mark.parametrize('filename', ['out.data', 'out.data.bz2'])
```
Comment (RMeli) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]:
```suggestion
@pytest.mark.parametrize('filename', ['out.data', 'out.data.bz2', 'out.data.gz'])
```
?

Comment (IAlibay) [reply testsuite/MDAnalysisTests/coordinates/test_lammps.py]: Oops forgot the one I was actually meant to fix 🙃

File: package/MDAnalysis/coordinates/CRD.py [hunk 2]
```diff
@@ -128,6 +128,9 @@ class CRDWriter(base.WriterBase):
     .. versionchanged:: 2.2.0
        CRD extended format can now be explicitly requested with the
        `extended` keyword
+    .. versionchanged:: 2.6.0
+       Files are now writen in `wt` mode, and keep extensions, allowing
```
Comment (RMeli) [inline package/MDAnalysis/coordinates/CRD.py]:
```suggestion
       Files are now written in `wt` mode, and keep extensions, allowing
```

File: package/MDAnalysis/coordinates/PDBQT.py [hunk 3]
```diff
@@ -195,6 +195,11 @@ class PDBQTWriter(base.WriterBase):
 
     .. _PDB: http://www.wwpdb.org/documentation/file-format-content/format32/v3.2.html
     .. _PDBQT: http://autodock.scripps.edu/faqs-help/faq/what-is-the-format-of-a-pdbqt-file
+
+
+    .. versionchanged:: 2.6.0
+       Files are now writen in `wt` mode, and keep extensions, allowing
```
Comment (RMeli) [inline package/MDAnalysis/coordinates/PDBQT.py]:
```suggestion
       Files are now written in `wt` mode, and keep extensions, allowing
```

File: package/MDAnalysis/coordinates/PQR.py [hunk 4]
```diff
@@ -186,7 +186,11 @@ class PQRWriter(base.WriterBase):
 
     The output format is similar to ``pdb2pqr --whitespace``.
 
+
     .. versionadded:: 0.9.0
+    .. versionchanged:: 2.6.0
+       Files are now writen in `wt` mode, and keep extensions, allowing
```
Comment (RMeli) [inline package/MDAnalysis/coordinates/PQR.py]:
```suggestion
       Files are now written in `wt` mode, and keep extensions, allowing
```

File: package/MDAnalysis/lib/util.py [hunk 5]
```diff
@@ -338,9 +338,10 @@ def anyopen(datasource, mode='rt', reset=True):
         a file (from :class:`file` or :func:`open`) or a stream (e.g. from
         :func:`urllib2.urlopen` or :class:`io.StringIO`)
     mode: {'r', 'w', 'a'} (optional)
-        Open in r(ead), w(rite) or a(ppen) mode. More complicated
-        modes ('r+', 'w+', ...) are not supported; only the first letter of
-        `mode` is used and thus any additional modifiers are silently ignored.
+        Open in r(ead), w(rite) or a(ppen) mode. This string is directly
```
Comment (RMeli) [inline package/MDAnalysis/lib/util.py]:
```suggestion
        Open in r(ead), w(rite) or a(ppend) mode. This string is directly
```

File: package/MDAnalysis/lib/util.py [hunk 6]
```diff
@@ -338,9 +338,10 @@ def anyopen(datasource, mode='rt', reset=True):
...
+        passed to the file opening handler (either Python's openfe, bz2, or
+        gzip). More complex mode are supported if the file opening handler
```
Comment (RMeli) [inline package/MDAnalysis/lib/util.py]:
```suggestion
        gzip). More complex modes are supported if the file opening handler
```

**Review by RMeli — APPROVED** [review]
Small docs comments, otherwise LGTM

(That "t" key... 😉)

**Pull request merged**